helpers/form/class.Form.php

```diff
@@ -617,6 +618,27 @@ public function hasFileUpload()
 
         return $returnValue;
     }
+    
+    public function hasAsyncFileUpload(): bool
+    {
+        foreach ($this->elements as $element) {
+            if ($element instanceof tao_helpers_form_elements_xhtml_AsyncFile) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public function checkFormInstance(): bool
```

Comment:
IMO check seems you are validating something, which is not the case. You just need to know if this is a form instance.

```suggestion
    public function isFormInstance(): bool
```

Reply:
Done